aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+The Validation Message Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__  is used with the ``asp-validation-for`` attribute on a HTML `span <https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span>`__ element.
+
+.. code-block:: HTML
+  
+  <span asp-validation-for="Email">
+  
+The Validation Message Tag Helper will generate the following HTML:
+
+.. code-block:: HTML
+
+    <span class="field-validation-valid" 
+      data-valmsg-for="Email" 
+      data-valmsg-replace="true">
+
+The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__  adds the `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__  ``data-valmsg-for="property"`` attribute to the ``<span>`` element, which attaches the validation error messages on the input field of the specified model property with ``<span>`` element. When a client side validation error occurs, `jQuery <https://jquery.com/>`__ displays the error message in the ``<span>`` element.
+
+You generally use the `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__  after an ``Input`` Tag Helper for the same property. Doing so displays any validation error messages near the input that caused the error.
+
+:Note: You must have a view with the correct JavaScript and `jQuery <https://jquery.com/>`__ 
+ script references in place for client side validation. See :doc:`/mvc/models/validation` for more information.
+ 
+The HTML Helper `@Html.ValidationMessageFor <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/HtmlHelperValidationExtensions/index.html>`__ provides an alternative to this Tag Helper.
+
+The Validation Summary Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `Validation Summary Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationSummaryTagHelper/index.html>`__  is used to display a summary of validation messages. The `asp-validation-summary`` attribute value can be any of the following:
```

Comment:
- Need to mention this tag helper targets `<div>` elements with the `asp-validation-summary` attribute.
- Looks like a backtick is missing before `asp-validation-summary`.
- Mention the `@Html.ValidationSummary()` equivalent.